var/spack/repos/builtin/packages/openssl/package.py

```diff
@@ -333,7 +335,18 @@ class Openssl(Package):  # Uses Fake Autotools, should subclass Package
         ),
     )
     variant("docs", default=False, description="Install docs and manpages")
-    variant("shared", default=False, description="Build shared library version")
+    variant(
+        "shared", default=True, description="Build shared library version", when="platform=linux"
+    )
+    variant(
+        "shared", default=True, description="Build shared library version", when="platform=darwin"
+    )
+    variant(
+        "shared",
+        default=False,
+        description="Build shared library version",
+        when="platform=windows",
+    )
```

Comment:
This is wrong, see the documentation on [overriding variants](https://spack.readthedocs.io/en/latest/packaging_guide.html#overriding-variants):

> When a variant is defined multiple times, whether in the same package file or in a subclass and a superclass, the last definition is used for all attributes except for the when clauses. The when clauses are accumulated through all invocations, and the variant is present on the spec if any of the accumulated conditions are satisfied.

The default is `False` on all platforms.

Reply:
Thanks, that makes sense.